aten/src/ATen/core/type.cpp

```diff
@@ -585,6 +585,22 @@ bool is_null_or_equal(c10::optional<T> a, c10::IntArrayRef b) {
   return !a.has_value() || a.value() == b;
 }
 
+
+
+ bool matchStrides(TensorType* ttp, const at::Tensor& t) {
+   TORCH_INTERNAL_ASSERT(t.is_contiguous());
+   auto ssize = *ttp->stride_properties().size();
+   const auto& t_strides = t.strides();
+   for (size_t i = 0; i < ssize; i++) {
+     const auto& stride = *ttp->stride_properties()[i];
```

Comment:
idk if it makes a difference but since we're micro-optimizing maybe hoist `*ttp->stride_properties()` out of the loop.  And actually it could be even better to do a ranged for loop over `*ttp->stride_properties()` but I'm unsure of that.

Reply:
i'll use the range loop.. that's an excellent idea.

Reply:
>  we're micro-optimizing

I hope we aren't. The reason why I want to avoid calling `computeStrides` since it a) constructs a bunch of nested objects (some of them are on heap) and b) does a sort with a lambda and c) uses a bunch of intermediate vectors in between . In this respect, `matchStrides` seems much cleaner. It compares strides directly without building an intermediate object for an input tensor.